<launch>
  <arg name="debug" default="false" />
  <arg name="output" default="log"/>
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix"
	   value="/usr/local/bin/gdb --directory=/usr/local/bin --ex run --args" />
  <arg name="octomap_topic" default="/octomap_binary"/>
  <arg name="hist_resolution" default="0.087222222" />
  <arg name="max_plan_range" default="3.0" />
  <arg name="map_resolution" default="0.15" />
  <arg name="const_a" default="10.0" />

  <arg name="visual_material" default="DarkGrey" />
  <arg name="rotors_description_dir" default="$(find px4)/Tools/sitl_gazebo/models/rotors_description" />
  <param name="robot_description" 
    command="$(find xacro)/xacro 
      $(arg rotors_description_dir)/urdf/iris.xacro 
      rotors_description_dir:=$(arg rotors_description_dir) 
      visual_material:=$(arg visual_material)
      --inorder" 
  />

  <!-- Start the vfh3d node -->
  <node name="vfh3d_node" pkg="vfh3d" type="vfh3d_node" output="$(arg output)" launch-prefix="$(arg launch_prefix)">
    <param name="octomap_topic" value="$(arg octomap_topic)" />
    <param name="hist_resolution" value="$(arg hist_resolution)" />
    <param name="map_resolution" value="$(arg map_resolution)" />
    <param name="max_plan_range" value="$(arg max_plan_range)" />
    <param name="const_a" value="$(arg const_a)" />
    <param name="pose_topic" value="/mavros/local_position/pose" />
    <param name="cmd_pose_topic" value="/mavros_moveit/local_position/cmd_pose" />
    <param name="target_vel_topic" value="/vfh3d/target_vel" />
    <param name="vehicle_safety_radius" value="0.2" />
    <param name="turning_radius_l" value="0.2" />
    <param name="turning_radius_r" value="0.2" />
  </node>
</launch>
